packages/hdwallet-tally/src/adapter.ts

```diff
@@ -0,0 +1,47 @@
+import detectEthereumProvider from "@metamask/detect-provider";
+import * as core from "@shapeshiftoss/hdwallet-core";
+import TallyOnboarding from "tallyho-onboarding";
+
+import { TallyHDWallet } from "./tally";
+
+export class TallyAdapter {
+  keyring: core.Keyring;
+
+  private constructor(keyring: core.Keyring) {
+    this.keyring = keyring;
+  }
+
+  public static useKeyring(keyring: core.Keyring) {
+    return new TallyAdapter(keyring);
+  }
+
+  public async initialize(): Promise<void> {
+    // no initialization needed
+  }
+
+  public async pairDevice(): Promise<TallyHDWallet> {
+    const provider: any = await detectEthereumProvider({ mustBeMetaMask: false, silent: true, timeout: 3000 });
```

Comment:
Should Tally detection depend on @metamask/detect-provider?

Reply:
No, this was something I thought about but didn't change. I did change the onboarding for the same reason. Ive remove this dependency and did it in a vanilla way as suggested.